### PR TITLE
fix bug with logout

### PR DESCRIPTION
**Pull request by Hat331**

### CODE REVIEW

File: src/scripts/StoreRestApi.js [hunk 1]
```diff
@@ -15,10 +16,18 @@ export default class StoreRestApi {
...
 	postInit() {
-		this.axiosInstance.interceptors.response.use(response => {
+		this.axiosInstance.interceptors.response.use(async response => {
+			console.log(response);
 			const contentType = response.headers['content-type'];
-			if (contentType !== 'application/json') {
-				const error = new Error(`Expected JSON response from API, but got ${contentType}`);
+			if (
+				contentType !== 'application/json' &&
+				response.request.responseURL.includes('auth')
```
Comment (mxsnq) [inline src/scripts/StoreRestApi.js:24]: move to talisman store

File: src/scripts/StoreRestApi.js [hunk 2]
```diff
@@ -15,10 +16,18 @@ export default class StoreRestApi {
...
+			) {
+				console.log('auth error');
```
Comment (mxsnq) [inline src/scripts/StoreRestApi.js:26]: duplicates error in promise

File: src/scripts/Controller.js [hunk 3]
```diff
@@ -18,6 +18,13 @@ import Translator from './Translator';
 
 export default class SitemapController {
 	constructor(store, templateDir) {
+		browser.runtime.onMessage.addListener(async request => {
+			if (request.authError) {
+				$('#confirm-action-modal').remove();
+				$('.modal-backdrop').remove();
+				return this.showAuthPage();
+			}
+		});
```
Comment (mxsnq) [inline src/scripts/Controller.js:27]: I'd move into init method

File: src/background/background.js [hunk 4]
```diff
@@ -16,7 +16,7 @@ config.loadConfiguration().then(() => {
 	console.log('initial configuration', config);
 	if (config.storageType === 'rest') {
 		store = new StoreRestApi(config, config.restUrl);
-		store.postInit();
+		store.setAxiosInterceptors();
```
Comment (mxsnq) [inline src/background/background.js:19]: here and on line 32: you already call it from StoreRestApi constructor

File: src/scripts/StoreRestApi.js [hunk 5]
```diff
@@ -12,13 +13,14 @@ export default class StoreRestApi {
...
+		this.setAxiosInterceptors();
 	}
 
-	postInit() {
+	setAxiosInterceptors() {
 		this.axiosInstance.interceptors.response.use(response => {
 			const contentType = response.headers['content-type'];
 			if (contentType !== 'application/json') {
-				const error = new Error(`Expected JSON response from API, but got ${contentType}`);
+				const error = new Error(`Authentication Error`);
```
Comment (mxsnq) [inline src/scripts/StoreRestApi.js:23]: not necessary auth error in this case

File: src/scripts/StoreTalismanApi.js [hunk 6]
```diff
@@ -10,8 +11,8 @@ export default class StoreTalismanApi extends StoreRestApi {
 
 	async initTalismanLogin(credentials) {
 		let bodyForm = new FormData();
-		let tLogin = credentials.username;
-		let tPassword = credentials.password;
+		const tLogin = credentials.username;
+		const tPassword = credentials.password;
```
Comment (mxsnq) [inline src/scripts/StoreTalismanApi.js:15]: add const to bodyForm too

File: src/scripts/StoreTalismanApi.js [hunk 7]
```diff
@@ -44,6 +42,23 @@ export default class StoreTalismanApi extends StoreRestApi {
...
+				response.request.responseURL.includes('auth')
+			) {
+				browser.runtime.sendMessage({
+					authError: true,
+				});
+				const error = new Error(`Authentication Error`);
+				return Promise.reject(error);
+			}
+			return response;
+		});
```
Comment (mxsnq) [inline src/scripts/StoreTalismanApi.js:59]: Maybe we should alert for any non-json responses in both storages? You allow such responses here unless they are not auth responses.

**Review by mxsnq — COMMENTED** [review]
please raise version in package.json

**Review by mxsnq — DISMISSED** [review]

**Review by Yatskov — APPROVED** [review]

**Pull request merged**